src/GafferScene/Shader.cpp

```diff
@@ -537,6 +617,122 @@ class Shader::NetworkBuilder
 					}
 				}
 			}
+			else if( (Gaffer::TypeId)parameter->typeId() == SplineffPlugTypeId )
+			{
+				addSplineParameterComponentConnections< SplineffPlug >( (const SplineffPlug*) parameter, parameterName, connections );
+			}
+			else if( (Gaffer::TypeId)parameter->typeId() == SplinefColor3fPlugTypeId )
+			{
+				addSplineParameterComponentConnections< SplinefColor3fPlug >( (const SplinefColor3fPlug*)parameter, parameterName, connections );
+			}
+			else if( (Gaffer::TypeId)parameter->typeId() == SplinefColor4fPlugTypeId )
+			{
+				addSplineParameterComponentConnections< SplinefColor4fPlug >( (const SplinefColor4fPlug*)parameter, parameterName, connections );
+			}
+		}
+
+		template< typename T >
+		void addSplineParameterComponentConnections( const T *parameter, const IECore::InternedString &parameterName, vector<IECoreScene::ShaderNetwork::Connection> &connections )
```

Comment:
Would be good to have some test coverage for this in `GafferSceneTest.ShaderTest`, where we're close to the code and can query `ShaderNetworks` directly.